lib/src/group_changes.dart

```diff
@@ -127,23 +151,27 @@ List<SummaryData> summarizePageData(
   final Map<String, int> hashIndex = Map.fromEntries(
       Iterable.generate(hashes.length, (i) => MapEntry(hashes[i], i)));
 
+  // New tests don't have a previous commit or result.
+  // for our heuristic, we group them together if they are within
+  // 10 commits of each other.
+  int previousHashIndex(Map<String, dynamic> change) {
+    if (change['previous_commit_hash'] == null) {
+      return hashIndex[change['commit_hash']] + 10;
+    } else {
+      return hashIndex[change['previous_commit_hash']];
+    }
+  }
+
   final resultsForTestAndChange =
-      Map<String, Map<String, List<Map<String, dynamic>>>>();
+      Map<String, Map<Result, List<Map<String, dynamic>>>>();
```

Comment:
This is a bit odd since 'Result' is more than just a key.

Reply:
The Result object is declared to be equal to any other Result object with the same key.  The only field not in the key, "matches", is a deterministic function of two fields that are in the key.

I could have left "key" out of Result, and declared operator== and hashCode directly in terms of the fields.  "key" is not currently used in anything other than hashCode and operator==.

Reply:
Removing key.  operator== checks all fields.